core/src/main/java/org/apache/spark/api/java/function/MapFunction.java

```diff
@@ -23,5 +23,5 @@
  * Base interface for a map function used in Dataset's map function.
  */
 public interface MapFunction<T, U> extends Serializable {
-  U call(T value) throws Exception;
```

Comment:
These function APIs will be implemented by users and called by us, why do we declare it to throw exceptions? It brings a lot of trouble when using it in codegen, but seems no benefit?  cc @rxin